arch/risc-v/src/common/supervisor/riscv_machine_trap.S

```diff
@@ -0,0 +1,164 @@
+/****************************************************************************
+ * arch/risc-v/src/common/supervisor/riscv_machine_trap.S
```

Comment:
Also all of the files in /supervisor/ folder that are related to the machine mode functionality (mcall.c, machine trap.S mscratch.c, mtimer.c, nx_start_trampoline) should be moved to its own folder ?